server/plugins/client/bundle.js

```diff
@@ -14,20 +28,27 @@ var parallel = require('async').parallel
  * and avoiding unneeded bundling with browserify saves a significant time.
  */
 function bundleClient (hoodieClientPath, bundleTargetPath, config, callback) {
-  parallel([
+  var plugins = [
+    path.resolve(config.paths && config.paths.applicationRoot || '', 'hoodie/client')
+  ].filter(checkModule)
+  var getPluginsModifiedTimes = plugins.map(function (pluginPath) {
+    return getModifiedTime.bind(null, requireResolve(pluginPath))
+  })
+
+  parallel(getPluginsModifiedTimes.concat([
     getModifiedTime.bind(null, hoodieClientPath),
     getModifiedTime.bind(null, bundleTargetPath)
-  ], function (error, results) {
+  ]), function (error, results) {
     /* istanbul ignore if */
     if (error) {
       return callback(error)
     }
 
-    var sourceTime = results[0]
-    var targetTime = results[1]
+    var sourceTime = Math.max.apply(null, results.slice(0, -1))
+    var targetTime = results[results.length - 1]
```

Comment:
> ```js
> var targetTime = results.pop()
> var sourceTime = Math.max.apply(Math, results)
> ```

D'oh! Yep this is more efficient. I think I was trying to getaway without mutating the argument, but it is throwaway data after all. I'll make this change.